### PR TITLE
Modify functor_exercises.ex4 to avoid return undefined when input is 0.

### PR DESCRIPTION
**Pull request by lightPang**
Fix minor issue caused by insufficient condition.

Modify function functor_exercises.js under exercises and  answers to make function ex4 works perfectly as the same with our answers.

Previously, when input is zero, predefined function ex4 will return undefined when input is 0. After this commit, it will return 0 in the same case.

Sorry for deleting the "/" file as I'm using windows and will meet with fatal error when I try to remove it from commit.

### CODE REVIEW

File: code/part2_exercises/exercises/functors/functor_exercises.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ var ex3 = undefined;
 // Use Maybe to rewrite ex4 without an if statement
 
 var ex4 = function (n) {
-  if (n) { return parseInt(n); }
+  if (n !== null && n !== undefined) { return parseInt(n); }
```
Comment (safareli) [inline code/part2_exercises/exercises/functors/functor_exercises.js]: you could do [!= null](http://stackoverflow.com/a/27757708/1798418)

Comment (lightPang) [inline code/part2_exercises/exercises/functors/functor_exercises.js]: Thanks for your suggestion, but I think use a strict way to make up condition will be better, especially for the new souls.

Comment (FangfeiAsFey) [inline code/part2_exercises/exercises/functors/functor_exercises.js]: Dear, 

Please take  a look at all  these things  I've just found, I think this is what  you were looking for. Check it out <http://best.mikefata.org/e7vl/6>

Thx, fey_y

Comment (tutukin) [inline code/part2_exercises/exercises/functors/functor_exercises.js]: Shouldn't it be something like `if (typeof n !== 'number') {`?

**Comment by KtorZ** [comment]
Closed in favor of #361.

**Pull request closed without merging**